log4j2-propagation/src/main/java/nl/talsmasoftware/context/log4j2/ThreadContextManager.java

```diff
@@ -0,0 +1,167 @@
+import nl.talsmasoftware.context.clearable.ClearableContextManager;
+
+/**
+ * Manager to propagate the Log4j 2 {@link ThreadContext} data from one thread to another.
+ * <p>
+ * As {@code ThreadContext} already manages its own threadlocal state,
+ * getting the active context is delegated to the {@code ThreadContext}.
+ * This means that closing the resulting context from {@link #getActiveContext()} will have
+ * no effect, because its data is not managed by this library. However, calling
+ * {@link #clear()} will clear the {@code ThreadContext} data of the current thread.<br>
+ * Methods of this manager may have no effect when the {@code ThreadContext} has been disabled
+ * (see <a href="https://logging.apache.org/log4j/2.x/manual/thread-context.html#Configuration">Log4j 2 manual</a>).
+ * <p>
+ * Initializing a new context through {@link #initializeNewContext(ThreadContextData)} will
+ * add the context data on top of the existing one, if any: {@code ThreadContext} stack values
+ * are pushed on top of the existing stack; map entries are added to the existing map, only
+ * replacing existing ones in case of a map key conflict.<br>
+ * Closing a context returned form {@link #initializeNewContext(ThreadContextData)} will reset
+ * the {@code ThreadContext} to the values it had before the context was created.<br>
+ * This means that closing nested contexts out-of-order will probably result in an undesirable state.<br>
+ * It is therefore strongly advised to use Java's {@code try-with-resources} statement to ensure proper
+ * closing when nesting {@code ThreadContext} contexts.
+ * <p>
+ * This manager and the contexts it creates are not suited to create {@code ThreadContext} 'scopes'
+ * within the same thread. Log4j 2's {@link CloseableThreadContext} should be used for that instead.
+ * <p>
+ * Log4j 2 supports making {@code ThreadContext} inheritable, i.e. to have it use
+ * {@link InheritableThreadLocal}. In some cases this might solve the problem of propagating
+ * context from one thread to another. However in all cases where threads are reused (such
+ * as thread pool executors) this will not work. It is therefore recommended to always
+ * prefer using this library to cover all use cases.
+ * <p>
+ * As with all manager implementations of this library there is usually no need to directly
+ * interact with the manager classes. Instead Java's {@link ServiceLoader} makes sure they
+ * are loaded as services. If an instance of this class is needed nonetheless it can be obtained
+ * through the field {@link #INSTANCE}.
+ *
+ * @see <a href="https://logging.apache.org/log4j/2.x/manual/thread-context.html">Log4j 2 Thread Context manual</a>
+ */
+public class ThreadContextManager implements ClearableContextManager<ThreadContextData> {
+    /**
+     * Singleton instance of this class.
+     */
+    public static final ThreadContextManager INSTANCE = new ThreadContextManager();
+
+    /**
+     * Returns the singleton instance.
```

Comment:
I think it does not hurt to mention this implementation detail; especially since the reason why this method exists is that there is a singleton instance. Otherwise the no-arg constructor would suffice for `ServiceLoader`.